django/db/models/query.py

```diff
@@ -798,6 +798,27 @@ def exists(self):
             return self.query.has_results(using=self.db)
         return bool(self._result_cache)
 
+    def contains(self, obj):
+        """
+        Check if object exists in queryset.
+
+        If the QuerySet is already fully cached, check if object is in cache.
+        """
+        if not issubclass(self._iterable_class, ModelIterable):
```

Comment:
I would prefer checking `_fields`
```suggestion
        if self._fields is not None:
```